### PR TITLE
Change the exit condition in the `user_liked` method

### PR DESCRIPTION
**Pull request by ilyailya**
Hi folks 👋 

I met a problem when I try to fetch user liked videos by using `user_liked` method. 
If user has less liked videos than a `count` parameter, then method would always return `[]`
and log `User's likes are most likely private`. 

I've changed the exit condition to address this issue. 

I've also updated the return statement in case of no videos in response.
Before the PR, the method would return empty list even if any subsequent request after the first one fails. 
After the change the method will return everything it had pulled in previous requests.

### CODE REVIEW

File: TikTokApi/tiktok.py [hunk 1]
```diff
@@ -847,12 +847,12 @@ def user_liked(self, userID, secUID, count=30, cursor=0, **kwargs) -> dict:
                 res["itemList"]
             except Exception:
                 logging.error("User's likes are most likely private")
-                return []
```
Comment (davidteather) [inline TikTokApi/tiktok.py:850]: 👍

File: TikTokApi/tiktok.py [hunk 2]
```diff
@@ -847,12 +847,12 @@ def user_liked(self, userID, secUID, count=30, cursor=0, **kwargs) -> dict:
...
+                return response
 
             for t in res.get("itemList", []):
                 response.append(t)
 
-            if not res.get("hasMore", False) and not first:
+            if not res.get("hasMore", False) or not first:
```
Comment (davidteather) [inline TikTokApi/tiktok.py:855]: This change causes the package to only return the first page of results, even if hasMore is true. I think the main improvement you saw was from the line 850 change. If you just revert the `or` back to `and` I'll merge :)

**Review by davidteather — CHANGES_REQUESTED** [review]
Line 855

**Comment by ilyailya** [comment]
Here you go @davidteather

**Review by davidteather — APPROVED** [review]
Thanks :)

**Pull request merged**